### PR TITLE
fix: auth Asana bug fixes

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of changes
- alias attributes bug fix
- mfa enabled functionality added 
- tsc global install error fix
- added a migration for headless update for old Apps
- added a migration test for the same
- Fixed unit tests after rebase

#### Description of how you validated changes
- manually Tested
- ran all e2e  test_suites for auth [working]

#### Checklist

- [X] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sachscode — APPROVED** [review]

File: packages/amplify-category-auth/src/commands/auth/import.ts [hunk 1]
```diff
@@ -1,15 +1,17 @@
 import { $TSContext } from 'amplify-cli-core';
 import { projectHasAuth } from '../../provider-utils/awscloudformation/utils/project-has-auth';
-import { supportedServices } from '../../provider-utils/supported-services';
-import { importResource } from '../../provider-utils/awscloudformation';
+const { getSupportedServices } = require('../../provider-utils/supported-services');
```
Comment (jhockett) [inline packages/amplify-category-auth/src/commands/auth/import.ts:3]: Can we use import syntax? `supported-services` is already in TS

File: packages/amplify-category-auth/src/commands/auth/import.ts [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 export const run = async (context: $TSContext) => {
   if (projectHasAuth(context)) {
     return;
   }
+  const servicesMetadata = getSupportedServices();
 
-  const serviceSelection = await context.amplify.serviceSelectionPrompt(context, category, supportedServices);
-  return importResource(context, serviceSelection, undefined);
+  const serviceSelection = await context.amplify.serviceSelectionPrompt(context, category, servicesMetadata);
+  const providerController = require(`../../provider-utils/${serviceSelection.providerName}`);
```
Comment (jhockett) [inline packages/amplify-category-auth/src/commands/auth/import.ts:14]:
```suggestion
  const providerController = await import(path.join('..', '..', 'provider-utils', serviceSelection.providerName));
```

Comment (jhockett) [reply packages/amplify-category-auth/src/commands/auth/import.ts:14]: `path` would have to be imported for this suggested change to work

File: packages/amplify-cli-core/src/cfnUtilities.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ export function readCFNTemplate(filePath: string, options: Partial<typeof defaul
     }
     throw new Error(`No CloudFormation template found at ${filePath}`);
   }
-
+  //TODO:  somthing wrong with this call , work fine with readFileSync()
```
Comment (jhockett) [inline packages/amplify-cli-core/src/cfnUtilities.ts:24]: nit: can this comment be removed now that it's using readFileSync()?

**Review by jhockett — APPROVED** [review]

**Pull request merged**